### PR TITLE
[Enhancement][RHEL/7] Update pam_faillock content to use and check for unlock_time=never

### LINKED ISSUE #1582: Update accounts_passwords_pam_faillock_unlock_time to support unlock_time=never
RHEL 7.3 now supports unlock_time=never, per downstream BZ:
https://bugzilla.redhat.com/show_bug.cgi?id=1273373

Need to update SSG XCCDF and OVAL to support ``unlock_time >= $value`` *and* ``unlock_time=never``

### PR DESCRIPTION
**Pull request by redhatrises**
- Fixes #1582

### CODE REVIEW

**Comment by redhatrises** [comment]
Should this also be changed in the OSPP profile as well?

File: shared/oval/accounts_passwords_pam_faillock_unlock_time.xml [hunk 1]
```diff
@@ -38,35 +38,35 @@
...
+    <ind:subexpression datatype="string" operation="equals" var_ref="var_accounts_passwords_pam_faillock_unlock_time" />
   </ind:textfilecontent54_state>
 
   <ind:textfilecontent54_state id="state_accounts_passwords_pam_faillock_unlock_time_password-auth" version="2">
-    <ind:subexpression datatype="int" operation="equals" var_ref="var_accounts_passwords_pam_faillock_unlock_time" />
+    <ind:subexpression datatype="string" operation="equals" var_ref="var_accounts_passwords_pam_faillock_unlock_time" />
   </ind:textfilecontent54_state>
 
-  <external_variable comment="number of failed login attempts allowed" datatype="int" id="var_accounts_passwords_pam_faillock_unlock_time" version="2" />
+  <external_variable comment="number of failed login attempts allowed" datatype="string" id="var_accounts_passwords_pam_faillock_unlock_time" version="2" />
```
Comment (shawndwells) [inline shared/oval/accounts_passwords_pam_faillock_unlock_time.xml]: I don't know how to handle it in code given the combination of string + integer values, but how will we handle:

- pass if unlock_time <= var_accounts_passwords_pam_faillock_unlock_time or unlock_time="never";
- fail if unlock_time >= var_accounts_passwords_pam_faillock_unlock_time and unlock_time != "never"

Comment (redhatrises) [inline shared/oval/accounts_passwords_pam_faillock_unlock_time.xml]: @shawndwells the way the OVAL is currently written even before this PR is that all integer values have to equal `var_accounts_passwords_pam_faillock_unlock_time`. Why do we need to have `<=` or `>=` when `var_accounts_passwords_pam_faillock_unlock_time` is expected to be set to some value in the OVAL?

Comment (radzy) [inline shared/oval/accounts_passwords_pam_faillock_unlock_time.xml]: @shawndwells : The combination could be handled with OR over two criteria.  Not sure if that's the best, but it should work.

**Comment by shawndwells** [comment]
On 11/28/16 7:57 PM, redhatrises wrote:
> @shawndwells <https://github.com/shawndwells> the way the OVAL is
> currently written even before this PR is that all integer values have
> to equal |var_accounts_passwords_pam_faillock_unlock_time|. Why do we
> need to have |<=| or |>=| when
> |var_accounts_passwords_pam_faillock_unlock_time| is expected to be
> set to some value in the OVAL?

I think the way we've written the OVAL check is wrong. And we never
fixed it. Ultimately, if the requirement is to lock accounts for 10
minutes and SysAdmin configures for 15min, the check should pass. If
configured for 5min, the system should fail.

Happy to merge this PR (because this *is* progress!) and open a ticket
to review sub/super compliance testing elsewhere.

**Comment by redhatrises** [comment]
> Happy to merge this PR (because this *is* progress!) and open a ticket to review sub/super compliance testing elsewhere.

@shawndwells please do both.

**Comment by shawndwells** [comment]
done. opened https://github.com/OpenSCAP/scap-security-guide/issues/1606

**Pull request merged**